python-threatexchange/threatexchange/cli/experimental_fetch.py

```diff
@@ -36,75 +44,102 @@ def init_argparse(cls, ap) -> None:
             type=int,
             help="Fetch updates that occured before this timestamp",
         )
-        ap.add_argument(
-            "--owner",
-            type=int,
-            help="Only fetch updates for indicators that the given app has a descriptor for",
-        )
         ap.add_argument(
             "--threat-types",
             nargs="+",
             help="Only fetch updates for indicators of the given type",
         )
         ap.add_argument(
-            "--additional-tags",
-            nargs="+",
-            help="Only fetch updates for indicators that have a descriptor with each of these tags",
+            "--page-size",
+            type=int,
+            help="The number of updates to fetch per request, defaults to 500",
+            default=500,
         )
 
     def __init__(
         self,
+        continuation: bool,
         start_time: int,
         stop_time: int,
-        owner: int,
         threat_types: t.List[str],
-        additional_tags: t.List[str],
+        page_size: int,
     ) -> None:
+        self.continuation = continuation
         self.start_time = start_time
         self.stop_time = stop_time
-        self.owner = owner
         self.threat_types = threat_types
-        self.additional_tags = additional_tags
-        self.signal_types_by_name = {
-            name: signal() for name, signal in meta.get_signal_types_by_name().items()
-        }
+        self.limit = page_size
         self.last_update_printed = 0
         self.counts = collections.Counter()
+        self.total_count = 0
+        self.deleted_count = 0
 
     def execute(self, dataset: Dataset) -> None:
-        # TODO: [Potential] Force full fetch if it has been 90 days since the last fetch.
-        self.start_time = (
-            dataset.get_indicator_checkpoint()
-            if self.start_time is None
-            else self.start_time
-        )
-        self.stop_time = int(time.time()) if self.stop_time is None else self.stop_time
-        dataset.load_indicator_cache(self.signal_types_by_name.values())
-        more_to_fetch = True
+        request_time = int(time.time())
+        privacy_group = dataset.config.privacy_groups[0]
+        self.indicator_store = dataset.get_indicator_store(privacy_group)
+        dataset.load_indicator_cache(self.indicator_store)
+        self.stop_time = request_time if self.stop_time is None else self.stop_time
         next_page = None
+        if self.continuation:
+            checkpoint = dataset.get_indicator_checkpoint(privacy_group)
+            self.start_time = checkpoint["last_stop_time"]
+            self.stop_time = request_time
+            self.threat_types = checkpoint["threat_types"]
```

Comment:
Can you print the params being used for the continuation?

Reply:
I can but this might get a little weird if the continuation is based on a url/cursor:
So the url would contain the original start time, stop time, threat types, limit, but the user would need to know how cursors work in order to understand that we are not actually fetching all data from start time to stop time.

The easier thing here would probably be if we are starting from a url/cursor then print out the corresponding url, if we are starting a new time-range based query print out the start time, stop time, threat types, and limit, thoughts?